src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -830,6 +847,47 @@ evalApp γ ctx e0 es et
       , f == "Language.Haskell.Liquid.ProofCombinators.pleUnfold"
       = arg
       | otherwise = e
+evalApp γ _ctx e0 es _et
+  | EVar f <- dropECst e0
+  , Just eq <- Map.lookup f (knAms γ)
+  = do
+    isHOAllowed <- gets hoFlag
+    if isHOAllowed then do
+    let expectedArgs = eqArgs eq
```

Comment:
Indentation is amiss. It should be nested at least two further spaces to the right.